src/modules/ProjectsModule/components/ProjectsList/ProjectsList.tsx

```diff
@@ -1,10 +1,179 @@
+        autoClose: 3000,
+        hideProgressBar: true,
+        pauseOnHover: false,
+      });
+    }
+  }
+
+    const formatDate = (dateString) => {
+    const date = new Date(dateString);
+    const month = date.getMonth() + 1;
+    const day = date.getDate();
+    const year = date.getFullYear();
+    return `${month}/${day}/${year}`;
+  };
+
+  const handleUpdate = (id,title,description) => {
+    setMode('update')
+    setTitle(title)
+    setDescription(description)
+    navigate(`/dashboard/projects-data/${id}`)
+  }
+
+  const getNameValue = (e) => {
+    getProjects(e.target.value)
+  }
+
+  useEffect(() => {
+  getProjects('')
+  }, [])
 
-const ProjectsList = () => {
   return (
-    <>
-     ProjectsList 
-    </>
-  )
+<>
+        <Modal show={showDelete} onHide={handleCloseDelete}>
+        <Modal.Header closeButton></Modal.Header>
+        <Modal.Body className="my-2 px-5 ">
+      <div className="text-center">
+      <img className="w-25" src={deleteTrash} alt="" />
+      <h4 className="py-3">Delete This Project ?</h4>
+      <p className="text-muted">
+        are you sure you want to delete this item ? if you are sure just click
+        on delete it
+      </p>
+      </div>
+          
+        </Modal.Body>
+        <Modal.Footer>
+          <Button variant="danger" onClick={deleteProjectItem}>
+            Delete This Item
+          </Button>
+        </Modal.Footer>
+      </Modal>
+
+    <div className='projectsList vh-100'>
+      <div className="projects-container bg-white px-3 d-flex justify-content-between py-4">
+        <h3>Projects</h3>
+        <Link onClick={setMode('create')} className='px-4 rounded-5' to={'/dashboard/projects-data'}> <i className='fa fa-plus pe-3'></i> Add New Project</Link>
+      </div>
+        <div className="projects-crud m-3  bg-white py-4 rounded-2">
+              <div className="projects-filteration px-3 ">
+          <input type="search" onChange={getNameValue} className='form-control w-25 rounded-5' placeholder='Search By Title' />
+        </div>
+        {projectsList.length ? (
+          <>
+        <div className="my-3">
+          <Table striped bordered hover className="text-center">
+            <thead>
+              <tr >
+                <th>ID</th>
+                <th>Title</th>
+                <th>description</th>
+                <th>Tasks</th>
+                <th>Creation Date</th>
+                <th>Actions</th>
+              </tr>
+            </thead>
+            <tbody>
+              {projectsList.map((ele) => 
+                <tr key={ele.id}>
+                  <td>{ele.id}</td>
+                  <td>{ele.title}</td>
+                  <td>{ele.description}</td>
+                  <td>{ele.tasks && ele.tasks[0]?.length}</td>
+                  <td>{ formatDate(ele.creationDate) }</td>
+                  <td>
+                    <i
+                      onClick={() => handleUpdate(ele.id,ele.title,ele.description)}
```

Comment:
Icons shouldn't be given `onClick`. This needs to be converted to icon buttons, and the onClick should be given to the button itself not the icon for better accessibility.